In [1]:
import json
import os

In [2]:
os.getcwd()

'C:\\Users\\shrey\\Downloads\\LTA'

In [3]:
f = open('C:/Users/shrey/Downloads/LTA/NTU_congestion_project/Data_examples/LTA_DataMall_SpeedBand/speed-2022-08-01/23-59-02.json', "r")
  
# Reading from file
speed_data = json.loads(f.read())

In [4]:
f = open ('C:/Users/shrey/Downloads/LTA/TrafficIncidents/1aug/Final_Accidents.json', "r")
  
# Reading from file
acci_data = json.loads(f.read())

In [5]:
len(acci_data)

4

In [6]:
acci_data[0]

{'Type': 'Accident',
 'Latitude': 1.3228381210778106,
 'Longitude': 103.8905865880709,
 'Message': '(1/8)7:43 Accident on PIE (towards Changi Airport) after Paya Lebar Rd with congestion till Thomson Rd. Avoid lanes 1 and 2.'}

In [7]:
(speed_data[59005]['Location'])

'1.4607500991003515 103.8104326729758 1.461077477013782 103.8102700220811'

In [8]:
# def findingloco(lati,longi):
#     for i in range(len(speed_data)):
#         k = [] #list to temporarily store the coordinates to compare
#         k = speed_data[i]['Location'].split()

In [9]:
for i in range(len(speed_data)):
        k = [] #list to temporarily store the coordinates to compare
        k = speed_data[i]['Location'].split()
#         trying to find the location of the accidents
        if str(1.3228381210778106) <= k[0] and str(1.3228381210778106) >= k[2] and str(103.8905865880709) <= k[1] and str(103.8905865880709) >=k[3]:
            print(i)
        

4623
49049


In [10]:
speed_data[49049]

{'LinkID': '104005395',
 'RoadName': 'PAN ISLAND EXPRESSWAY',
 'RoadCategory': 'A',
 'SpeedBand': 8,
 'MinimumSpeed': '70',
 'MaximumSpeed': '999',
 'Location': '1.3228613329448404 103.8907263045645 1.3226128387058194 103.88924256381655'}

# Plotting a temporal relationship for no. of links congested nearby accident location which is on LinkdID: 104005395 , Time: 7:43, Location:'Latitude': 1.3228381210778106, 'Longitude': 103.8905865880709,

In [11]:
f = open('C:/Users/shrey/Downloads/LTA/NTU_congestion_project/Data_examples/LTA_DataMall_SpeedBand/speed-2022-08-01/23-59-02.json', "r")
  
# Reading from file
speed_data1 = json.loads(f.read())

In [12]:
speed_data1[49049]

{'LinkID': '104005395',
 'RoadName': 'PAN ISLAND EXPRESSWAY',
 'RoadCategory': 'A',
 'SpeedBand': 8,
 'MinimumSpeed': '70',
 'MaximumSpeed': '999',
 'Location': '1.3228613329448404 103.8907263045645 1.3226128387058194 103.88924256381655'}

# To get nearby links of the accident

In [13]:
Linkids_nearby1 = []
for i in range(len(speed_data1)):
    k = []
    k = speed_data1[i]['Location'].split()
    if k[0][:5] == str(1.322) and k[1][:5] == str(103.8):
        Linkids_nearby1.append(speed_data1[i])
    
    

In [14]:
type(Linkids_nearby1[1]['SpeedBand'])

int

In [15]:
# now counting the number of links with 'SpeedBand' = 1 or 2 as that means speed is less than 20km/hr
n = 0
for i in range(len(Linkids_nearby1)):
    if Linkids_nearby1[i]['SpeedBand'] == 2 or Linkids_nearby1[i]['SpeedBand'] == 1:
        n = n+1
print(n)

14


In [16]:
n

14

In [17]:
g = open('C:/Users/shrey/Downloads/LTA/NTU_congestion_project/Data_examples/LTA_DataMall_SpeedBand/speed-2022-08-01/23-59-02.json', "r")
  
# Reading from file
speed_data0 = json.loads(g.read())

In [18]:
Linkids_nearby0 = []
for i in range(len(speed_data0)):
    k = []
    k = speed_data0[i]['Location'].split()
    if k[0][:5] == str(1.322) and k[1][:5] == str(103.8):
        Linkids_nearby0.append(speed_data0[i])

In [19]:
(Linkids_nearby0)

[{'LinkID': '103000162',
  'RoadName': 'AKYAB ROAD',
  'RoadCategory': 'E',
  'SpeedBand': 5,
  'MinimumSpeed': '40',
  'MaximumSpeed': '49',
  'Location': '1.3223315936500188 103.84782453064221 1.323225112339397 103.84825791705197'},
 {'LinkID': '103001244',
  'RoadName': 'BASSEIN ROAD',
  'RoadCategory': 'F',
  'SpeedBand': 8,
  'MinimumSpeed': '70',
  'MaximumSpeed': '999',
  'Location': '1.3229138443765627 103.84886573248687 1.3230710117237383 103.84920749817688'},
 {'LinkID': '103001250',
  'RoadName': 'BASSEIN ROAD',
  'RoadCategory': 'E',
  'SpeedBand': 4,
  'MinimumSpeed': '30',
  'MaximumSpeed': '39',
  'Location': '1.3229138443765627 103.84886573248687 1.323225112339397 103.84825791705197'},
 {'LinkID': '103001758',
  'RoadName': 'SERANGOON ROAD',
  'RoadCategory': 'B',
  'SpeedBand': 2,
  'MinimumSpeed': '10',
  'MaximumSpeed': '19',
  'Location': '1.322378321650752 103.86377061045651 1.3227460150283434 103.8632888377743'},
 {'LinkID': '103001759',
  'RoadName': 'SERANGOON R

In [20]:
k = 0
for i in range(len(Linkids_nearby0)):
    if Linkids_nearby0[i]['RoadCategory'] == 'F':
         k = k+1
print(k)    

15


In [21]:
def count_of_congested_links(Linkids_nearby):
    k = 0
    for i in range(len(Linkids_nearby)):
        if Linkids_nearby[i]['SpeedBand'] == 2 or Linkids_nearby[i]['SpeedBand'] == 1:
            k = k+1
    return k

In [22]:
# def linkids_nearby(d):
#     r = []
#     for i in range(len)

In [23]:
count_of_congested_links(Linkids_nearby0)

14

In [24]:
Linkids_nearbyneg1 = []
for i in range(len(speed_data_neg1)):
    k = []
    k = speed_data_neg1[i]['Location'].split()
    if k[0][:5] == str(1.322) and k[1][:5] == str(103.8):
        Linkids_nearbyneg1.append(speed_data_neg1[i])

NameError: name 'speed_data_neg1' is not defined

In [ ]:
len(Linkids_nearbyneg1)

In [ ]:
count_of_congested_links(Linkids_nearbyneg1)